src/rvsdg/from_egglog.rs

```diff
@@ -0,0 +1,231 @@
+        let Expr::Call(func, args) = vec_vec else {
+        panic!("Expected a VVO, got {vec_vec}")
+      };
+        assert_eq!(func.as_str(), "VVO");
+        assert_eq!(args.len(), 1);
+        let vec_vec = &args[0];
+        vec_map(vec_vec, |vec| Self::expr_to_vec_operand(vec, bodies))
+    }
+
+    fn egglog_expr_to_body(body: &Expr, bodies: &mut Vec<RvsdgBody>) -> Id {
+        use Expr::*;
+        if let Call(func, args) = body {
+            let body = match (func.as_str(), &args.as_slice()) {
+                ("PureOp", [expr]) => RvsdgBody::BasicOp(Self::egglog_expr_to_expr(expr, bodies)),
+                ("Gamma", [pred, inputs, outputs]) => {
+                    let pred = Self::egglog_expr_to_operand(pred, bodies);
+                    let inputs = Self::expr_to_vec_operand(inputs, bodies);
+                    let outputs = Self::expr_to_vec_vec_operand(outputs, bodies);
+                    RvsdgBody::Gamma {
+                        pred,
+                        inputs,
+                        outputs,
+                    }
+                }
+                ("Theta", [pred, inputs, outputs]) => {
+                    let pred = Self::egglog_expr_to_operand(pred, bodies);
+                    let inputs = Self::expr_to_vec_operand(inputs, bodies);
+                    let outputs = Self::expr_to_vec_operand(outputs, bodies);
+                    RvsdgBody::Theta {
+                        pred,
+                        inputs,
+                        outputs,
+                    }
+                }
+                _ => panic!("expect an operand, got {body}"),
+            };
+            bodies.push(body);
+            bodies.len() - 1
+        } else {
+            panic!("expect an operand, got {body}")
+        }
+    }
+
+    fn egglog_expr_to_expr(expr: &Expr, bodies: &mut Vec<RvsdgBody>) -> RvsdgExpr<Operand> {
+        use egglog::ast::Literal;
+        if let Expr::Call(func, args) = expr {
+            match (func.as_str(), &args.as_slice()) {
+                ("Call", [ty, Expr::Lit(Literal::String(ident)), args]) => {
+                    let args = vec_map(args, |e| Self::egglog_expr_to_operand(e, bodies));
+                    // TODO: this is imprecise, we don't know if the number of outputs is 1 or 2.
+                    RvsdgExpr::Call(
+                        Identifier::Name(ident.to_string()),
+                        args,
+                        1,
+                        Some(Self::egglog_expr_to_ty(ty)),
+                    )
+                }
+                ("Const", [ty, _const_op, lit]) => RvsdgExpr::Const(
+                    // todo remove the const op from the encoding because it is always ConstOps::Const
+                    ConstOps::Const,
+                    Self::egglog_expr_to_literal(lit),
+                    Self::egglog_expr_to_ty(ty),
+                ),
+                ("PRINT", [opr1, opr2]) => {
```

Comment:
I think bril supports more than one arg to print (state edge or no), see https://github.com/oflatt/eggcc/blob/main/tests/brils/failing/char/char.bril

Reply:
yeah we just support one for now, we can fix in the future